### ISSUE TITLE
[Feature Request] Output forces on granular walls.

### ISSUE THREAD

**Issue by vikaskok** [lammps/lammps#2085]
**Summary**
It would be useful to obtain forces acing on granular walls as an output of "fix wall/gran command". Currently we cannot access the forces from this fix, unlike some other fix commands, like say "fix freeze".
From documentation of fix wall/gran command: "No global or per-atom quantities are stored by this fix for access by various output commands".

**Detailed Description**
Forces acting on bounding walls will help in determining if the simulation box "end-walls" are affecting the dynamics in the interior of the box. The information of forces acting on end walls will help in choosing a big enough simulation box.
One can also build walls using atoms by employing the fix freeze command for the atoms lying in the regions near simbox walls. We can also obtain total forces on the group of atoms on which fix freeze is applied. However, we can issue only one fix freeze command. So all the walls will have to be grouped together and we obtain total forces on ALL walls. However, if we wish to obtain forces on individual walls (left, right, top, etc.) it would be helpful to have forces output from "fix wall/gran".

**Further Information, Files, and Links**
There has been a short relevant discussion on this in e-mail forums in 2011: https://lammps.sandia.gov/threads/msg20643.html

**Comment by rbberger** [comment]
Looks like the docs are outdated. The implementation does support the `store_contacts` option, which defines a peratom property. Try this (not sure if I got the right indices):

```
fix zwall all wall/gran granular hertz/material  ... store_contacts

# contact force stored for each atom, reduce it to total force on wall
compute wforce all reduce sum f_zwall[2] f_zwall[3] f_zwall[4]

thermo_style custom step cpu atoms c_wforce[1] c_wforce[2] c_wforce[3]
```

**Comment by vikaskok** [comment]
Thank you very much for your prompt responses.
@rbberger : I carried out a few test runs with the store_contacts option enabled for the fix wall/gran command.
I would like to share with you the following observations:
1. Regarding indices. (Using the fix zwall example in your comment)
f_zwall[1] - A vector containing atom IDs which have come into contact with zwall.
f_zwall[2,3,4] - fx,fy,fz exerted on the wall respectively by corresponding atom (corresponding means relevant entry in f_zwall[1] vector)  which has come into contact with the wall.
f_zwall[5,6,7] - x,y,z coordinates of the corresponding atom which has come into contact with the wall.
f_zwall[8] - Radius of the atom which has come into contact with the wall.

2. Once an atom touches the wall, its ID is appended to the vector f_zwall[1]. However, when that atom detaches from the wall, its ID is **NOT** deleted from the f_zwall[1] vector. 
3. As long as a particle is interacting with the wall, the  forces exerted by that atom on the wall are updated in f_zwall[2,3,4] vectors. However, when the particle detaches from the wall, the last computed particle-wall force values are continued to be written to in f_zwall[2,3,4] vectors for the subsequent time steps. This is perhaps due to issue discussed in point 2. above.
4. I suspect, in a simulation containing large number of atoms, the f_zwall[1] vector keeps bulging (atom IDs keep getting appended) because once a particle comes into contact with the wall, albeit for a short period of time, it is continued to retain in f_zwall[1]. As a consequence, the wall force information is not correct. I have checked this by inserting one particle into a simbox and allowing it to bounce off from a wall. The wall forces are initially zero, change during the contact duration and continue to retain the force last exerted by the particle before loosing contact. So, we will see that even after the particle detaches and there is no particle in contact with the wall, forces on the wall (some/all entries of f_zwall[2,3,4]) will be non-zero.

Suggested edit:
In my opinion, point 2. above leads to such peculiar behaviour. Once the particle-wall contact is lost, the corresponding atom ID should be deleted from f_zwall[1] vector. The rest of the force information would then naturally get rectified.

Thanks again, for your time.

My LAMMPS version : lammps-stable_7Aug2019
